superflore/generators/bitbake/gen_packages.py

```diff
@@ -49,22 +49,58 @@ def regenerate_pkg(
     component_name = yoctoRecipe.convert_to_oe_name(
         distro.release_packages[pkg].repository_name)
     recipe = yoctoRecipe.convert_to_oe_name(pkg)
-    # check for an existing recipe
-    prefix = '{0}/meta-ros{1}-{2}/generated-recipes/{3}/{4}'.format(
-        repo_dir,
+    # check for an existing recipe which was removed by clean_ros_recipe_dirs
+    prefix = 'meta-ros{0}-{1}/generated-recipes/*/{2}_*.bb'.format(
         yoctoRecipe._get_ros_version(distro.name),
         distro.name,
-        component_name,
-        recipe,
+        recipe
     )
-    existing = glob.glob('{}_*.bb'.format(prefix))
+    existing = overlay.repo.git.status('--porcelain', '--', prefix)
+    if existing:
+        # The git status --porcelain output will look like this:
+        # D  meta-ros2-eloquent/generated-recipes/variants/ros-base_0.8.3-1.bb
+        # we want just the path with filename
+        if len(existing.split('\n')) > 1:
+            warn('More than 1 recipe was output by "git status --porcelain '
+                 'meta-ros{0}-{1}/generated-recipes/*/{2}_*.bb": "{3}"'
+                 .format(
+                     yoctoRecipe._get_ros_version(distro.name),
+                     distro.name,
+                     recipe,
+                     existing))
+        if existing.split()[0] != 'D':
+            err('Unexpected output from "git status --porcelain '
+                'meta-ros{0}-{1}/generated-recipes/*/{2}_*.bb": "{3}"'
+                .format(
+                    yoctoRecipe._get_ros_version(distro.name),
+                    distro.name,
+                    recipe,
+                    existing))
+
+        existing = existing.split()[1]
```

Comment:
If more than one recipe what output by `git status --porcelain`, is it always true that the first one is the one to process?

Reply:
Currently it's just used to show in commit message, I don't think it's important which one is used (as long as there was an warning about it and the commit message doesn't say that the component is new, we might sort them based on version and then choose the one with highest version - because that's probably the one bitbake was building before, but that seems unnecessary for commit message in cases like this which shouldn't ever happen again.